### DISCUSSION TITLE
TypeError: Got unsupported ScalarType BFloat16

### DISCUSSION THREAD

**Discussion by adil-ahmed** [coqui-ai/TTS#2811] [General Q&A] [answered]
Dear TTS GitHub Community,

I hope this message finds you well. I am a newcomer to the field of Text-to-Speech (TTS) and currently working on training the VITS (Variational Inference Text-to-Speech) model using the free version of Google Colab. I am eager to understand the process and generate the necessary checkpoints to explore the capabilities of the model further.

However, I have encountered several challenges during the training process, and despite my efforts, I have been unable to resolve them on my own. As a result, I am seeking your expert guidance and assistance to overcome these hurdles.

Here are the steps I have taken so far:

1. I downloaded the VCTK dataset from the following link: https://github.com/coqui-ai/TTS/blob/dev/recipes/vctk/download_vctk.sh and uploaded it to my Google Drive.
2. I made necessary modifications to the train_bits.py script according to my requirements and uploaded it to Google Drive as well.
3. I mounted the Google Drive in Google Colab and attempted to run the VITS model for training. Unfortunately, the process consistently results in errors, and I haven't been able to generate any checkpoints successfully.

Since I am new to this area, I might be missing some crucial steps or misconfiguring the training process. I would sincerely appreciate your valuable insights and recommendations to resolve the issues I'm facing and make progress with the VITS TTS model training. If possible, I can share the relevant code snippets or error logs for your reference. Please let me know the best way to provide this information. Thank you so much for your time and support. I am looking forward to your guidance and expertise.

**Error**

> TRAINING (2023-07-27 22:28:26) 
/usr/local/lib/python3.10/dist-packages/torch/functional.py:641: UserWarning: stft with return_complex=False is deprecated. In a future pytorch release, stft will return complex tensors for all inputs, and return_complex=False will raise an error.
Note: you can still call torch.view_as_real on the complex output to recover the old return format. (Triggered internally at ../aten/src/ATen/native/SpectralOps.cpp:862.)
  return _VF.stft(input, n_fft, hop_length, win_length, window,  # type: ignore[attr-defined]

   --> TIME: 2023-07-28 00:14:30 -- STEP: 0/54 -- GLOBAL_STEP: 0
     | > loss_disc: 5.965904235839844  (5.965904235839844)
     | > loss_disc_real_0: 1.0246418714523315  (1.0246418714523315)
     | > loss_disc_real_1: 0.9844479560852051  (0.9844479560852051)
     | > loss_disc_real_2: 0.9797080755233765  (0.9797080755233765)
     | > loss_disc_real_3: 1.0096845626831055  (1.0096845626831055)
     | > loss_disc_real_4: 0.9764307737350464  (0.9764307737350464)
     | > loss_disc_real_5: 0.9904793500900269  (0.9904793500900269)
     | > loss_0: 5.965904235839844  (5.965904235839844)
     | > grad_norm_0: tensor(6.7017)  (tensor(6.7017))
     | > loss_gen: 4.672023296356201  (4.672023296356201)
     | > loss_kl: 186.6778564453125  (186.6778564453125)
     | > loss_feat: 0.2575872242450714  (0.2575872242450714)
     | > loss_mel: 98.01626586914062  (98.01626586914062)
     | > loss_duration: 2.399357318878174  (2.399357318878174)
     | > loss_1: 292.0231018066406  (292.0231018066406)
     | > grad_norm_1: tensor(1531.9602)  (tensor(1531.9602))
     | > current_lr_0: 0.0002 
     | > current_lr_1: 0.0002 
     | > step_time: 6342.521  (6342.520982027054)
     | > loader_time: 21.2059  (21.205898761749268)

 ! Run is kept in /content/drive/MyDrive/All/Personal/UBishops/ModelTraining/VITS/trainingResult/vits_vctk-July-27-2023_10+27PM-0000000
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 1805, in fit
    self._fit()
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 1757, in _fit
    self.train_epoch()
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 1486, in train_epoch
    outputs, _ = self.train_step(batch, batch_num_steps, cur_step, loader_start_time)
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 1455, in train_step
    self.update_training_dashboard_logger(batch=batch, outputs=outputs)
  File "/usr/local/lib/python3.10/dist-packages/trainer/utils/distributed.py", line 35, in wrapped_fn
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 1941, in update_training_dashboard_logger
    self.model.train_log(
  File "/usr/local/lib/python3.10/dist-packages/TTS/tts/models/vits.py", line 1366, in train_log
    figures, audios = self._log(self.ap, batch, outputs, "train")
  File "/usr/local/lib/python3.10/dist-packages/TTS/tts/models/vits.py", line 1336, in _log
    figures = plot_results(y_hat, y, ap, name_prefix)
  File "/usr/local/lib/python3.10/dist-packages/TTS/vocoder/utils/generic_utils.py", line 48, in plot_results
    y_hat = y_hat[0].squeeze().detach().cpu().numpy()
TypeError: Got unsupported ScalarType BFloat16

**Comment by Cherchercher** [comment]
Same, I'm running from a GPU on collab as context

**Comment by erogol** [comment, marked as answer]
if you are using mixed_precision training make sure you set the `precision` parameter correctly. You can use Bfloat16 only on systems that support it.
